### PR TITLE
Add sorting to rm

### PR DESCRIPTION
**Pull request by plehocky**
| Q                        | A 
| ------------------------ | ---
| Fixed Issue?             | `FD-551` 
| Tests Added + Pass?      | Yes
| Any Dependency Changes?  |

https://frinxhelpdesk.atlassian.net/jira/software/c/projects/FD/boards/77?modal=detail&selectedIssue=FD-551

test im RM -> pools page, click on either NAME or DEALOCATION SAFETY PERIOD in table head to sort

### CODE REVIEW

File: packages/frinx-resource-manager/src/nested-pools-detail-page/nested-pools-detail-page.tsx [hunk 1]
```diff
@@ -144,6 +146,11 @@ const NestedPoolsDetailPage: VoidFunctionComponent = () => {
     setSearchText(searchName);
   };
 
+  const handleSort = (field: 'name' | 'dealocationSafetyPeriod') => {
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/nested-pools-detail-page/nested-pools-detail-page.tsx:149]: use types provided by generated graphql types not your own, for future compatibility

File: packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx [hunk 2]
```diff
@@ -20,19 +20,22 @@ import {
...
 import PoolsTable from './pools-table';
 import SearchFilterPoolsBar from '../../components/search-filter-pools-bar';
 
-type InputValues = { [key: string]: string };
-
 type PoolsFilter =
   | {
-      resources: InputValues | null;
-      resourceType: string | null;
+      resources: { [key: string]: string | null | undefined } | null | undefined;
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx:29]: why did you delete the defined type and rewritten it in here?

File: packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx [hunk 3]
```diff
@@ -20,19 +20,22 @@ import {
...
+      resourceType: string | undefined | null;
     }
-  | Record<string, string>;
+  | null
+  | undefined;
+
+export type SortBy = {
+  field: 'name' | 'dealocationSafetyPeriod';
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx:36]: as mentioned in the first comment

**Review by MarcoMruz — CHANGES_REQUESTED** [review]

File: packages/frinx-resource-manager/src/pages/ipam-page/ipam-page.tsx [hunk 4]
```diff
@@ -124,7 +126,7 @@ const IpamPoolPage: VoidFunctionComponent = () => {
   const [poolsFilter, setPoolsFilter] = useState<PoolsFilter>({ resources: null, resourceType: null });
   const [isIpv4, setIsIpv4] = useState<boolean>(true);
   const [selectedResourceType, setSelectedResourceType] = useState<string>('');
-  const [allocatedResources, setAllocatedResources] = useState({});
+  const [allocatedResources, setAllocatedResources] = useState<{ [key: string]: string }>({});
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/pages/ipam-page/ipam-page.tsx:129]: use Record<string, string>

File: packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx [hunk 5]
```diff
@@ -20,19 +20,25 @@ import {
...
   GetResourceTypesQuery,
   GetResourceTypesQueryVariables,
-  Scalars,
+  ResourcePoolOrderField,
 } from '../../__generated__/graphql';
 import PoolsTable from './pools-table';
 import SearchFilterPoolsBar from '../../components/search-filter-pools-bar';
 
-type InputValues = { [key: string]: string };
+type InputValues = { [key: string]: string | null | undefined };
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/pages/pools-page/pools-page.tsx:28]: same as above

File: packages/frinx-resource-manager/src/pages/pools-page/pools-table.tsx [hunk 6]
```diff
@@ -39,13 +40,16 @@ type Pools = {
...
   PoolType: PoolType;
+  DealocationSafetyPeriod?: number;
   ResourceType: Record<string, string>;
   Resources: Resource[];
   Tags: Omit<TagType, 'Pools'>[];
   allocatedResources?: Record<string, number | string> | null;
 };
 
 type Props = {
+  onSort?: (sortKey: 'name' | 'dealocationSafetyPeriod') => void;
```
Comment (MarcoMruz) [inline packages/frinx-resource-manager/src/pages/pools-page/pools-table.tsx:51]: use generated type by graphql API, to future proof

**Review by MarcoMruz — CHANGES_REQUESTED** [review]

**Review by MarcoMruz — APPROVED** [review]

**Review by Paulooze — APPROVED** [review]

**Pull request merged**